### PR TITLE
Adds concept tls generation for client certs

### PR DESCRIPTION
**Pull request by chuckbutler**

### CODE REVIEW

File: reactive/tls.py [hunk 1]
```diff
@@ -89,6 +89,27 @@ def create_csr(tls):
...
+        cn = 'client'
+        hookenv.log('Creating the CSR for {}'.format(cn))
+        with chdir('easy-rsa/easyrsa3'):
+            sans = get_sans()
+            path_name = "client"
+            gen_req = './easyrsa --batch --req-cn={0} --subject-alt-name={1} ' \
+            'build-client-full {2} nopass 2>&1'.format(cn, sans, path_name)
+            check_call(split(gen_req))
+    db = unitdata.kv()
+    with open('easy-rsa/easyrsa3/pki/issued/client.crt') as f:
```
Comment (mbruzek) [inline reactive/tls.py]: If you did the read inside the chdir block the path would not be so long.

File: reactive/tls.py [hunk 2]
```diff
@@ -89,6 +89,27 @@ def create_csr(tls):
...
+        client_certificate = f.read()
+    db.set('tls.client.certificate', client_certificate)
+    set_state('client certificate signed')
```
Comment (mbruzek) [inline reactive/tls.py]: available has been the convention but the states are arbitrary.

Comment (chuckbutler) [inline reactive/tls.py]: this follows the nomenclature of 'server certificate signed'

File: reactive/tls.py (hunk 1 above)
Comment (chuckbutler) [inline reactive/tls.py]: good point :thumbsup:

**Comment by mbruzek** [comment]
I saw what you did here, and I liked it!  This proposal has many good things that my proposal does not.  Like relying on reactive states to generate the client and has a client method. I think we should combine the approaches.

**Pull request closed without merging**